### PR TITLE
Support to install addon with CRD

### PR DESCRIPTION
**Pull request by LinuxSuRen**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:
See the details from the document.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:

```
None
```

### Does this PR introduce a user-facing change??

```release-note
Support to install addon with CRD
```

### CODE REVIEW

File: controllers/addon/operator_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
+package addon
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (JohnNiang) [inline controllers/addon/operator_controller_test.go:8]: I'd prefer to give it a name to `metav1`.

File: docs/addon.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+| [ks-releaser](https://github.com/kubesphere-sigs/ks-releaser/) | Help to release a project which especially has multiple git repositories. |
+| [Argo CD](https://github.com/argoproj/argo-cd/)                | Declarative continuous deployment for Kubernetes.                         |
+
+## How to use?
+
+The `Addon Controller` is optional, please add the flag `--enabled-controllers addon=true` into the controller command line.
+For instance:
+
+```yaml
+
```
Comment (JohnNiang) [inline docs/addon.md:22]: May miss something here.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: controllers/addon/operator_controller_test.go (hunk 1 above)
Comment (LinuxSuRen) [reply controllers/addon/operator_controller_test.go]: good catch!

**Review by JohnNiang — APPROVED** [review]
/lgtm

**Pull request merged**